src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2732,12 +2731,22 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 		g_assert (fsig->hasthis && fsig->param_count == 2 && fsig->params [0]->type == MONO_TYPE_I4 && fsig->params [1]->type == MONO_TYPE_R4);
 
 		gboolean indirect = FALSE;
+		int elems = 4, index = args [1]->inst_c0;
 		int dreg = load_simd_vreg (cfg, cmethod, args [0], &indirect);
 
-		MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, args [1]->dreg, len);
-		MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "ArgumentOutOfRangeException");
-		ins = emit_simd_ins (cfg, klass, OP_XINSERT_R4, dreg, args [2]->dreg);
-		ins->sreg3 = args [1]->dreg;
+		// Bounds check only if the index is out of range
+		if (index < 0 || index >= elems) {
+			MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, args [1]->dreg, len);
+			MONO_EMIT_NEW_COND_EXC (cfg, GE_UN, "ArgumentOutOfRangeException");
+		}
+
+		if (COMPILE_LLVM (cfg)) {
```

Comment:
I believe that there is something that we could do to simplify this code path. Please open an issue to track it. You could say something like "Merge LLVM and non-llvm code path in emit_vector_2_3_4" in the issue.

Reply:
https://github.com/dotnet/runtime/issues/89616